### ISSUE TITLE
Importing methods

### ISSUE THREAD

**Issue by flying-sheep** [rticulate/import#16]
Hi!

I’m a fan of https://github.com/thomasp85/patchwork

It defines functions like `-.ggplot` which I’d like to import. I need to use `import:::from` for this, which seems unintuitive: when the patchwork package is attached, I can use those functions, so why would I need to use private accession syntax?

**Comment by smbache** [comment]
It's because they're not exported (https://github.com/thomasp85/patchwork/blob/master/NAMESPACE). S3methods are a bit special like that. Not sure exactly why that is. `import` deliberately makes it different when you want to import non-exported things. Not sure how to handle s3methods.. interesting.

**Comment by smbache** [comment]
```
get(".__NAMESPACE__.", envir=asNamespace("patchwork"))[["S3methods"]]
```
reveals the available methods, whereas 
```
ls(get(".__S3MethodsTable__.", envir=asNamespace("patchwork")))
```
does not.

Perhaps S3 deserves their own function `import::S3from(...)` which could check the `S3methods` entry in the namespace to validate the import call, and then internally do the `:::` call.

**Comment by torfason** [comment]
This did not make it into the 1.2.0 release. I'm not sure about the cost-benefit ration given that it seems to be fixable with one extra `:` if I understand it correctly. I suggest we put this on ice unless a stronger rationale emerges for fixing this (even better if such a rationale were accompanied with a nice PR of course :smiley:).

**Comment by J-Moravec** [comment]
I just find about this issue.

I wrote a new method for a base generic functions and as everything else, I just did `import::from("src/code.r", "generic.class").

To my surprise, when calling `generic(object)`, the `generic.class` was not dispatched, despite `generic.class(object)` could be called directly.

Changing from `import::from` to `import::here` does fix the issue. Although it's also importing the code directly, which I am not sure I am happy about.
